google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolStressTest.java

```diff
@@ -226,22 +229,26 @@ public void stressTest() throws Exception {
     }
     pool =
         SessionPool.createPool(
-            builder.build(), new TestExecutorFactory(), mockSpanner.getSessionClient(db), clock);
+            builder.build(),
+            new TestExecutorFactory(),
+            mockSpanner.getSessionClient(db),
+            clock,
+            Position.RANDOM);
     pool.idleSessionRemovedListener =
         pooled -> {
           String name = pooled.getName();
-          synchronized (lock) {
-            sessions.remove(name);
-            return null;
-          }
+          // We do not take the test lock here, as we already hold the session pool lock. Taking the
```

Comment:
This change is unrelated to the session pool re-balancing, but it fixes a potential deadlock in this test. The deadlock was caused by the fact that the session pool maintenance task has taken the `SessionPool#lock` when it comes here. Trying to also take the `lock` of this test case could cause a deadlock if one of the tests in this file had already taken this lock, and then tried to execute an action on the session pool that would also require the `SessionPool#lock`.